### ISSUE TITLE
Deno: Folder write permission is spuriously requested when loading a package

### ISSUE THREAD

**Issue by JanPokorny** [pyodide/pyodide#4624]
## 🐛 Bug

When running Pyodide in Deno, write permission to the package folder is requested when loading a package even for packages that don't need to be downloaded. Given the Deno output (pasted below), the offending operation is `mkdir`, thus I suspect there is some `mkdir` operation happening that is effectively no-op when the folder already exists.

### To Reproduce

```
import pyodideModule from "npm:pyodide/pyodide.js"

await pyodideModule.loadPyodide({ packages: ["numpy"] })
```

Run with: `deno run --node-modules-dir --allow-read=. test.ts`

Output (path redacted):

```
┌ ⚠️  Deno requests write access to "/[...]/node_modules/.deno/pyodide@0.25.0/node_modules/pyodide/".
├ Requested by `Deno.mkdir()` API.
├ Run again with --allow-write to bypass this prompt.
└ Allow? [y/n/A] (y = yes, allow; n = no, deny; A = allow all write permissions)
```

### Expected behavior

This should work without the write permission -- read access is all the code above needs. The package `numpy` is a part of the default distribution.

### Environment

- Pyodide Version: 0.25.0
- Browser version: Deno 1.41.3
- Any other relevant information:

### Additional context

Workaround is giving `deno` the write permission for the folder, but at the same time denying it using filesystem permissions.

**Comment by ryanking13** [comment]
> when loading a package even for packages that don't need to be downloaded.

> The package numpy is a part of the default distribution.

No. This is not true. NumPy is not included in the default distribution. Actually, we don't include any wheels in the npm package to reduce the size of the npm Pyodide package. Check the `node_modules/pyodide` directory.

Instead, we download packages on the fly in Node.js and store them in the node_modules directory. ([Code Pointer](https://github.com/pyodide/pyodide/blob/163a46e3ca5128a5c25b79bb6b4db76b7f5d9b30/src/js/load-package.ts#L290-L300))

This is not an ideal situation, but we don't want to host hundreds of megabytes of packages in a single npm package.

**Comment by JanPokorny** [comment]
@ryanking13 Oh, right, don't know why I thought `numpy` was included, it's indeed cached in the `node_modules` folder.

But the point still stands -- even when no changes are needed, write access is requested, which does not make sense.

![image](https://github.com/pyodide/pyodide/assets/4580066/1ad83b98-e4ca-4d14-90a1-e0120c880ebb)

I think the culprit is some needless mkdir, given that the request is for `Deno.mkdir`. Most likely there's some code that creates a folder in order to ensure that the folder exists, which prompts for the write permission.

Eg. something like this:

```
create_folder(folder)
do_something_with_folder(folder)
```

...should be changed into this

```
if not exists(folder):
  create_folder(folder)
do_something_with_folder(folder)
```

**Comment by JanPokorny** [comment]
@ryanking13 I found the offending line: https://github.com/pyodide/pyodide/blob/654204775bf6f5bc66239048137fffb1d1a51804/src/js/load-package.ts#L258

Patching it out fixes the problem:
![image](https://github.com/pyodide/pyodide/assets/4580066/9e7b7e62-08e0-415c-8ce7-27bae9d0daf0)

Adding a check here if the folder exists would prevent Deno for needlessly asking for permission. Sorry for not sending a full PR, I'm not very proficient with Node.js

**Comment by ryanking13** [comment]
Oh, I see. That makes sense. Thanks for the detailed information @JanPokorny! I'll see if I can fix it by checking the existence of the path first.

**Comment by ryanking13** [comment]
Fixed by @whitphx #4738.

**Comment by JanPokorny** [comment]
That's great, thank you @whitphx !
